overlord/snapstate/aliases_test.go

```diff
@@ -890,7 +890,164 @@ func (s *snapmgrTestSuite) TestAliasMatrixTotalUndoRunThrough(c *C) {
 		}
 		// start with an easier-to-read error if this fails:
 		c.Assert(s.fakeBackend.ops.Ops(), DeepEquals, expected.Ops(), comm)
-		c.Assert(s.fakeBackend.ops, DeepEquals, expected, comm)
+		c.Check(s.fakeBackend.ops, DeepEquals, expected, comm)
+
+		var allAliases map[string]map[string]string
+		err = s.state.Get("aliases", &allAliases)
+		c.Assert(err, IsNil)
+		if scenario.beforeStatus != "" {
+			c.Check(allAliases, DeepEquals, map[string]map[string]string{
+				"alias-snap": {scenAlias: scenario.beforeStatus},
+			}, comm)
+		} else {
+			c.Check(allAliases, HasLen, 0, comm)
+		}
+
+		s.fakeBackend.ops = nil
+	}
+}
+
+func (s *snapmgrTestSuite) TestDisabledSnapResetAliasesRunThrough(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	snapstate.Set(s.state, "alias-snap", &snapstate.SnapState{
+		Sequence: []*snap.SideInfo{
+			{RealName: "alias-snap", Revision: snap.R(11)},
+		},
+		Current: snap.R(11),
+		Active:  false,
+	})
+
+	// alias1 is a non auto-alias
+	// alias5 is an auto-alias
+	// alias1gone is a non auto-alias and doesn't have an entry in the current snap revision anymore
+	// alias5gone is an auto-alias and doesn't have an entry in the current snap revision anymore
+	snapstate.AutoAliases = func(st *state.State, info *snap.Info) ([]string, error) {
+		c.Check(info.Name(), Equals, "alias-snap")
+		return []string{"alias5", "alias5lost"}, nil
+	}
+
+	defer s.snapmgr.Stop()
+	for _, scenario := range statusesMatrix {
+		if scenario.action != "reset" {
+			// only reset
```

Comment:
Maybe worth to expand the command a little bit? Like: `we only test "reset" for disabled snaps as the other actions do not make sense for disabled snaps` or something like this?